libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioAdapter.cs

```diff
@@ -142,13 +148,16 @@ public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot
                     // create a conversation reference
                     using (var context = new TurnContext(this, activity))
                     {
+                        context.TurnState.Add("httpStatus", "200");
+
                         await RunPipelineAsync(context, bot.OnTurnAsync, cancellationToken).ConfigureAwait(false);
 
-                        response.StatusCode = 200;
+                        response.StatusCode = Convert.ToInt32(context.TurnState.Get<string>("httpStatus"));
```

Comment:
I think that instead of using `Convert.ToInt32(context.TurnState.Get<string>("httpStatus"))` we should be using `(int)HttpStatusCode.OK` as suggested by Gabo